astropy/utils/data.py

```diff
@@ -1124,47 +1148,61 @@ def is_url_in_cache(url_key):
-    except OSError as e:
-        msg = 'Remote data cache could not be accessed due to '
-        estr = '' if len(e.args) < 1 else (': ' + str(e))
-        warn(CacheMissingWarning(msg + e.__class__.__name__ + estr))
-        return False
+    with _cache() as (dldir, url2hash):
+        return url_key in url2hash
 
-    with shelve.open(urlmapfn) as url2hash:
-        if url_key in url2hash:
-            return True
-    return False
 
+def cache_total_size():
+    """Return the total size in bytes of all files in the cache."""
+    with _cache() as (dldir, url2hash):
+        return sum(os.path.getsize(os.path.join(dldir, h))
+                   for h in url2hash.values())
 
-def _do_download_files_in_parallel(args):
-    return download_file(*args)
 
+def _do_download_files_in_parallel(kwargs):
+    with astropy.config.paths.set_temp_config(kwargs.pop("temp_config")):
+        with astropy.config.paths.set_temp_cache(kwargs.pop("temp_cache")):
+            return download_file(**kwargs)
 
-def download_files_in_parallel(urls, cache=True, show_progress=True,
-                               timeout=None):
-    """
-    Downloads multiple files in parallel from the given URLs.  Blocks until
-    all files have downloaded.  The result is a list of local file paths
-    corresponding to the given urls.
+
+def download_files_in_parallel(urls,
+                               cache="update",
+                               show_progress=True,
+                               timeout=None,
+                               sources=None,
+                               multiprocessing_start_method=None):
+    """Download multiple files in parallel from the given URLs.
+
+    Blocks until all files have downloaded.  The result is a list of
+    local file paths corresponding to the given urls.
 
     Parameters
     ----------
     urls : list of str
         The URLs to retrieve.
 
-    cache : bool, optional
-        Whether to use the cache (default is `True`).
+    cache : bool or "update", optional
+        Whether to use the cache (default is `True`). If "update",
+        check the remote URLs to see if new data is available but then
+        store the result in cache.
+
+        .. versionchanged:: 4.0
+            The default was changed to ``"update"`` and setting it to
+            ``False`` will print a Warning and set it to ``"update"`` again,
+            because the function will not work properly without cache. Using
+            ``True`` will work as expected.
 
         .. versionchanged:: 3.0
-            The default was changed to ``True`` and setting it to ``False`` will
-            print a Warning and set it to ``True`` again, because the function
-            will not work properly without cache.
+            The default was changed to ``True`` and setting it to ``False``
+            will print a Warning and set it to ``True`` again, because the
+            function will not work properly without cache.
```

Comment:
I really wasn't sure what to do here. If people are upgrading from pre-3.0 versions, cache="update" gives them more or less what they wanted, while cache=True doesn't check whether the resource has changed. If they are updating from post-3.0 versions, they might be counting on the opposite behaviour. My reasoning was that knowing there had been a change, these folks would have put cache=True explicitly. And anyway they called a function "download_files_in_parallel", they shouldn't be shocked if files are downloaded. But I'm not sure a second change is the best way to handle this.

Reply:
I don't have the numbers of whether more people will upgrade from 2.x vs 3.x. In fact, the default here is different from default in `download_file`, but no one has complained, so maybe this is not even a widely used function. I would say just keep it as you have written unless @MSeifert04 has strong opinions, given that he changed the default for this function back in #6671.